### ISSUE TITLE
Change the user-kubeconfig name

### ISSUE THREAD

**Issue by felipeweb** [kubernetes-sigs/cluster-api-provider-aws#1951]
/kind feature

To be compatible with clusterctl get kubeconfig we need to change the kubeconfigs name. more details here https://github.com/kubernetes-sigs/cluster-api/issues/3645

- [cluster-name]-user-kubeconfig to [cluster-name]-kubeconfig
- [cluster-name]-kubeconfig to  [cluster-name]-internal-kubeconfig

**Comment by ncdc** [comment]
Are you talking about the kubeconfig Secret? If so, it would be very difficult to rename it. It is a breaking contract change. There are lots of consumers using the current naming conventions.

We could consider something like adding a label to the Secret we want `clusterctl get kubeconfig` to retrieve (this would be something to discuss in the cluster-api repo).

cc @vincepri

**Comment by vincepri** [comment]
This is interesting, who's creating the "user" secret from above in this case? Is this something very specific to the EKS provider?

`clusterctl get kubeconfig` could provide the option to get a user specific secret, although I'd like to see a proposal in Cluster API to understand how we'd create the secret and with what permissions. The aim is to provide a consistent user experience across all providers

**Comment by felipeweb** [comment]
@vincepri it is specific for EKS https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/docs/eks.md

**Comment by vincepri** [comment]
Got it, I'd suggest to use some extra documentation in the CAPI book or here and provide a kubectl command for now, we can open an upstream issue to track a larger story that would apply to all providers for v1alpha4 or later, how does that sound?

**Comment by richardcase** [comment]
@vincepri @felipeweb - we need the 2 kubeconfigs for the EKS side of the provider. The kubeconfig that a user needs to use to connect uses the `aws-iam-authenticator` or `aws cli` binary to generate the token and this caused a problem from capi when it used the kubeconfig as the binary didn't doesn't exist in the capi images so we created. So we decided to create 2 separate kubeconfigs and document this in the EKS specific docs whilst the feature is experimental....it has the kubectl command in that.

Sounds great that this situation for providers be solved upstream in capi in future versions...i just created https://github.com/kubernetes-sigs/cluster-api/issues/3661
